ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerImpl.java

```diff
@@ -289,7 +289,7 @@ public void onPackagerStatusFetched(final boolean packagerIsRunning) {
                     new Runnable() {
                       @Override
                       public void run() {
-                        if (packagerIsRunning) {
+                        if (packagerIsRunning || !mDevSupportManager.hasUpToDateJSBundleInCache()) {
```

Comment:
@astreet  I did this for #4739 so that, when the packager is not running and there is no update-to-date bundle, it would try to load a existent bundle and throw a confusing redbox message( the `else` part)